version: "3.8"

services:

  source_postgres:
    image: postgres:15.5
    ports:
      - "5433:5432"
    networks:
      - elt_network
    environment:
      POSTGRES_DB: source_db
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: secret
    volumes:
      - ./db_init/init.sql:/docker-entrypoint-initdb.d/init.sql
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      retries: 5
      timeout: 5s


  destination_postgres:
    image: postgres:15.5
    ports:
      - "5434:5432"
    networks:
      - elt_network
    environment:
      POSTGRES_DB: destination_db
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: secret


  elt_script:
    build:
      context: ./elt_script             
      dockerfile: Dockerfile           
    command: [ "python", "elt_script.py" ]  
    networks:
      - elt_network
    depends_on:
      source_postgres:
        condition: service_healthy


networks:
  elt_network:
    driver: bridge
